### ISSUE TITLE
Mature SR2ML frontend input structure

### ISSUE THREAD

**Issue by schelian** [Globalhelpforall/qv-ng-poc#12]

**Comment by schelian** [comment]
Tyler, do you need Aswin's help with CORS headers?  Aswin mentioned that the issue was fixed with the latest code, not the latest docker image.

**Comment by tylerstern0** [comment]
I have been unable to deploy the site yesterday and today via the command line; I think the changes made have made the previous instructions for running irrelevant. @aswinjose89 Do you have any updated info on running the code to deploy the frontend? I have tried using `ng serve` and `npm run hmr` and with either one, http://localhost:4200/ shows "Cannot GET" and I'm getting the error
`Property 'api_url' does not exist on type '{ SERVER_ORIGIN: string; production: boolean; useHash: boolean; hmr: boolean; }'.
39       `${environment.api_url}${path}`,
`
in the terminal.

**Comment by tylerstern0** [comment]
Disregard previous comment; ng serve is working now (npm run hmr is not still). I think yesterday my angular cli version was in some way whack, since I have to install it anew every time I use gilgamesh.

**Comment by tylerstern0** [comment]
CORS headers issue is indeed resolved.

**Comment by schelian** [comment]
So we can close this?  Do you still need to install things a new each time?

Forgot to tag @aswinjose89, @mikevasan98 in these issues.

**Comment by tylerstern0** [comment]
I don't think this issue describes the CORS headers bug so much as it does additional form fields, etc, so i would prefer we left it open until I have finished those. The CORS headers bug is definitely resolved, though. 

If I am not using the docker, I still need to reinstall certain libraries, but I think that is a permissions issue on gilgamesh rather than related to our system (and now I'm doing it correctly and can make a simple shell script to do it).

**Comment by schelian** [comment]
Notes 2022.03.18:

nodejs and npm (then angular) downloaded as tar.  Reset the PATH each time?!

Building w/o docker: ~5 minutes
Building w/ docker: 1 minute
"docker cp" the files from the host to the machine
restart angular command

DCV permissions may be different from SSH.  Please contact Sameer/Erik.

Other fields to add, 50% done.

O2 is null error.  If can't resolve please include Aswin by email.

**Comment by schelian** [comment]
Notes 2022.03.22:

ssh -Y works as a workaround for DCV
O2 is null error resolved.
Aswin sent instructions on how to build the docker; Tyler to try.
Frontend refreshes.
All fields but variables (distributions and samplers are up).
Tyler to update instructions for Suhas to try

**Comment by schelian** [comment]
Notes 2022.03.25:

Tyler checking connectivity.

Aswin to change raven-server a param in server run-docker.sh -- Want >1 person to run the backend at the same time.  Port name can be appended to image name.

Suhas to try on Voltar.

**Comment by schelian** [comment]
O2 error still there.  Tyler to try workaround with docker tar files.  Suhas can test these tar files.
